server/channels/api4/channel.go

```diff
@@ -89,6 +90,16 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if channel.TeamId == "" {
+		c.SetInvalidParamWithDetails("team_id", i18n.T("api.channel.create_channel.missing_team_id.error"))
+		return
+	}
+
+	if channel.DisplayName == "" {
+		c.SetInvalidParamWithDetails("display_name", i18n.T("api.channel.create_channel.missing_display_name.error"))
+		return
+	}
```

Comment:
What does the API currently return if `DisplayName` is empty? Is this a breaking change?

Reply:
It will still successfully create the channel. It might be a breaking change

Referring to [api docs](https://api.mattermost.com/#tag/channels/operation/CreateChannel)
Display Name should be required. 

Should I remove the validation to be safe or leave it there?

Reply:
If the API docs point out that display name is required, I thinkwe are good to enforce this

Reply:
Still if it's a breaking change in functionality we should at least raise it in the release notes.